src/ParametersHandler/tests/ParametersHandlerTest.cpp

```diff
@@ -176,14 +345,14 @@ TEST_CASE("Get parameters")
 
     SECTION("Print content")
     {
-        std::cout << "Parameters: " << *parameterHandler << std::endl;
+        std::cout << "Parameters: " << parameterHandler->toString() << std::endl;
     }
 
     SECTION("Set from object")
     {
         std::unordered_map<std::string, std::any> object;
         object["value"] = std::make_any<int>(10);
-        parameterHandler->set(object);
+        static_cast<BasicImplementation*>(parameterHandler.get())->set(object);
```

Comment:
I would prefer to avoid doing downcasting in the test. It gives a dangerous example. I would prefer changing line https://github.com/dic-iit/bipedal-locomotion-controllers/blob/a75238a837125c818b7ba5bee13c481d9b9d4c8c/src/ParametersHandler/tests/ParametersHandlerTest.cpp#L279

to 
```c++
 std::shared_ptr<BasicImplementation> originalHandler = std::make_shared<BasicImplementation>(); 
 IParametersHandler::shared_ptr parameterHandler = originalHandler; 
 ```
and access ``originalHandler`` here, while using the upcasted pointer everywhere else. This comment applies also to the ``YarpImplementation``.

Reply:
Done in https://github.com/dic-iit/bipedal-locomotion-controllers/pull/28/commits/3ee5e59c0906992c465d17d1afb8a451e6797783

Reply:
Can you do the same also in the ``YarpImplementationTest``?

Reply:
Yep

Reply:
> Can you do the same also in the `YarpImplementationTest`? (I mean, avoid the explicit downcasting in the test)

Done in 4d4c08835c59c74ca716c7f3a3670fa10005a4d7